pkg/broker/ingress/multi_topic_decouple_sink_test.go

```diff
@@ -0,0 +1,211 @@
+package ingress
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/cloudevents/sdk-go/v2/client/test"
+	cecontext "github.com/cloudevents/sdk-go/v2/context"
+	"github.com/cloudevents/sdk-go/v2/event"
+	"github.com/cloudevents/sdk-go/v2/protocol"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/knative-gcp/pkg/broker/config"
+	"github.com/google/knative-gcp/pkg/broker/config/memory"
+)
+
+type brokerTestCase struct {
+	ns          string
+	broker      string
+	topic       string
+	clientErrFn func(client *fakePubsubClient)
+	wantErr     bool
+}
+
+func TestMultiTopicDecoupleSink(t *testing.T) {
+	tests := []struct {
+		name         string
+		brokerConfig *config.TargetsConfig
+		cases        []brokerTestCase
```

Comment:
Similar to above, I would probably define the struct inline, unless you plan to use it outside this function.

Reply:
No I don't need it outside. However it seems that when I define `cases` I have to repeat the struct definition again like so:
```
	tests := []struct {
		name         string
		brokerConfig *config.TargetsConfig
		cases        []struct{
			ns          string
			broker      string
			topic       string
			clientErrFn func(client *fakePubsubClient)
			wantErr     bool
		}
	}{
		{
			name: "happy path single broker",
			brokerConfig: &config.TargetsConfig{
				Brokers: map[string]*config.Broker{
					"test_ns_1/test_broker_1": {DecoupleQueue: &config.Queue{Topic: "test_topic_1"},},
				},
			},
			cases: []struct{
				ns          string
				broker      string
				topic       string
				clientErrFn func(client *fakePubsubClient)
				wantErr     bool
			}{
				{
					ns:     "test_ns_1",
					broker: "test_broker_1",
					topic:  "test_topic_1",
				},
			},
		},
}
```

Let me know if there is a more concise way to do it. I will be very happy to change to that  :)

Reply:
Good point, that is quite ugly. How about defining the struct in the function?

```golang
func TestMultiTopicDecoupleSink(t *testing.T) {
	type brokerTestCase struct{
		ns          string
		broker      string
		topic       string
		clientErrFn func(client *fakePubsubClient)
		wantErr     bool
	}
	tests := []struct {
		name         string
		brokerConfig *config.TargetsConfig
		cases        []brokerTestCase
	}{
		{
			name: "happy path single broker",
			brokerConfig: &config.TargetsConfig{
				Brokers: map[string]*config.Broker{
					"test_ns_1/test_broker_1": {DecoupleQueue: &config.Queue{Topic: "test_topic_1"},},
				},
			},
			cases: []brokerTestCase{
				{
					ns:     "test_ns_1",
					broker: "test_broker_1",
					topic:  "test_topic_1",
				},
			},
		},
	}
...
}
```

Reply:
oh yes I can certainly do this